include/revng/Pipeline/Context.h

```diff
@@ -120,11 +123,47 @@ class Context {
 
 public:
   llvm::Error storeToDisk(llvm::StringRef Path) const {
-    return Globals.storeToDisk(Path);
+    if (auto Error = Globals.storeToDisk(Path); Error)
+      return Error;
+
+    return llvm::Error::success();
   }
+
   llvm::Error loadFromDisk(llvm::StringRef Path) {
-    return Globals.loadFromDisk(Path);
+    if (auto Error = Globals.loadFromDisk(*this, Path); Error)
+      return Error;
+
+    return llvm::Error::success();
+  }
+
+public:
+  void getTargetsAssociatedToPaths(const TupleTreePath &Path,
+                                   InvalidationMap &Out) const {
+    for (auto &Name : Globals.getGlobalNames()) {
+      llvm::cantFail(Globals.get(Name))->getTargetsAssociatedToPaths(Path, Out);
+    }
   }
```

Comment:
I understand why you might want `Global::getTargetsAssociatedToPaths` to take an `InvalidationMap` as an out argument, as opposed to returning it, but I think `Context::getTargetsAssociatedToPaths` can return it.

Reply:
i would prefer not too because sometimes this is invoked very intensively for a lot of paths different paths, and the result are all accumulated into the same Out invalidation map